File: cgi/env.c
#include <stdio.h>

int main(int ac, char **av, char **envp)
{
	if (ac > 1)
		(void)av;
	if (!envp)
		return 1;
	printf("Status: 200 OK\n");
	printf("Content-Type: text/plain\n\n");
	while (*envp) {
		printf("%s\n", *envp);
		++envp;
	}
	return (0);
}

File: cgi/env.py
#! /usr/bin/python3
import os

html = "<html><body><h2>Environment:</h2><br>"
for param in os.environ.keys():
    html += f"<p><b>{param}</b>: {os.environ[param]}</p>"
print("Content-Length: " + str(len(html)))
print("Content-Type: text/html\n")
print(html)
